packages/gitbook/src/lib/api.ts

```diff
@@ -87,21 +87,41 @@ const cacheTtl_1day = {
     ttl: 24 * 60 * 60,
 };
 
-const apiSyncStorage = new AsyncLocalStorage<GitBookAPI>();
+const apiSyncStorage = new AsyncLocalStorage<GitBookAPIWithContextKey>();
 
 export const DEFAULT_API_ENDPOINT = process.env.GITBOOK_API_URL ?? 'https://api.gitbook.com';
 
+export class GitBookAPIWithContextKey extends GitBookAPI {
```

Comment:
Do we need a class for this compared to just extending the type (`GitBookAPI & { comtextKey: string; }`)?

Reply:
Unless I'm missing something, yes I think. Because typescript would complain that returning `{ ...gitbookAPIClient, contextKey }` would not match `GitBookAPI & { comtextKey: string; }` type.

The other approach I considered was to do `Object.assign(gitbookAPIClient, { contextKey });` but I wasn't sure about it as it breaks the encapulation of `contextKey` prop by making it writeable and not read only prop anymore.

Can't think of any other way to get around it.

Reply:
Hum 🤔  it just feels ugly to create a huge class just for this, with such a name.

Maybe we solve it with:

```ts
type GitBookAPIContext = {
   client: GitBookAPI;
   contextKey: string | undefined;
}
```

and we just need to replace `api()` by `api().client`

It's probably cleaner